Changelog 192 (#180)
intellij-rust/intellij-rust.github.io: _posts/2023-04-10-changelog-192.markdown

---
layout: post
title: "IntelliJ Rust Changelog #192"

> date: 2023-04-10 13:00:00 +0300

Reviewer: ```suggestion
date: 2023-04-11 13:00:00 +0300
```

---

## New Features

* Detect new compiler errors:

  * [#10285] More than one explicit lifetime bound was used on a trait object [E0226]

  * [#10283] A union expression does not have exactly one field [E0784]

  * [#10268] Unsafe code was used outside an unsafe function or block [E0133] when accessing union field outside unsafe block

  * [#9375] Warn about unused label [unused_labels]

> * [#10281] Support `Rebuild ...` and `Clean` actions. You can access then from the `Build` menu

Reviewer: Maybe add a screenshot (like in PR)?

## Fixes

* [#10301] Fix expansion of macros that contain reserved keywords (e.g. `do`) in their body

* [#10295] Don't suggest a project refresh after creating a file in a nested dir in `tests` dir

* [#10266] Fix code analysis inside functions under `sqlx::test` macro

> * [#10265] Don't show "Conditionally disabled code" in the `Problems` view

Reviewer: Maybe add link for `Problems` view?
```suggestion
* [#10265] Don't show "Conditionally disabled code" in the [`Problems`](https://www.jetbrains.com/help/clion/problems-tool-window.html) view
```

---
layout: post
title: "IntelliJ Rust Changelog #192"
date: 2023-04-11 13:00:00 +0300
---

## New Features

* Detect new compiler errors:

  * [#10285] More than one explicit lifetime bound was used on a trait object [E0226]

  * [#10283] A union expression does not have exactly one field [E0784]

  * [#10268] Unsafe code was used outside an unsafe function or block [E0133] when accessing union field outside unsafe block

  * [#9375] Warn about unused label [unused_labels]

> * [#10281] Support `Rebuild ...` and `Clean` actions. You can access then from the `Build` menu

Reviewer 2: We definitely need to specify that it works only in CLion

## Fixes

* [#10301] Fix expansion of macros that contain reserved keywords (e.g. `do`) in their body

> * [#10295] Don't suggest a project refresh after creating a file in a nested dir in `tests` dir

Reviewer 2: ```suggestion
* [#10295] Don't suggest a project model reloading after creating a file in a nested dir in `tests` dir
```

* [#10266] Fix code analysis inside functions under `sqlx::test` macro

* [#10265] Don't show "Conditionally disabled code" in the [`Problems`] view

## Internal Improvements

> * [#10286] Initial implementation of MIR

Reviewer 2: I'd add a link for `MIR`
